config/v1/types_infrastructure_test.go

```diff
@@ -0,0 +1,33 @@
+package v1
+
+import (
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"testing"
+)
+
+const infraCRDFilePath =  "0000_10_config-operator_01_infrastructure.crd.yaml"
+
+func TestInfrastructureStatusDefault(t *testing.T) {
+	infraCRDBytes, err := ioutil.ReadFile(infraCRDFilePath)
+	if err != nil {
+		t.Fatalf("failed to read infrastructure CRD file %q: %v", infraCRDFilePath, err)
+	}
+
+	var infraCRD map[string]interface{}
+	if err := yaml.Unmarshal(infraCRDBytes, &infraCRD); err != nil {
+		t.Fatalf("failed to unmarshal infra CRD: %v", err)
+	}
+	infraCRDSpec := infraCRD["spec"].(map[interface{}]interface{})
+	infraCRDVersions := infraCRDSpec["versions"].([]interface{})
+	for _, v := range infraCRDVersions {
+		infraCRDVersion := v.(map[interface{}]interface{})
+		versionSchema := infraCRDVersion["schema"].(map[interface{}]interface{})
+		openAPIV3Schema := versionSchema["openAPIV3Schema"].(map[interface{}]interface{})
+		properties := openAPIV3Schema["properties"].(map[interface{}]interface{})
+		status := properties["status"].(map[interface{}]interface{})
```

Comment:
you can simplify with `unstructured.NestedField(infraCRDVersion, "schema", "openAPIV3Schema", "properties", "status")`

Reply:
I found this issue - https://github.com/go-yaml/yaml/issues/139, so at least for "gopkg.in/yaml.v2" I can not use unstructured, because it expected as input map[string]interface{} and I have map[interface{}]interface{}  and once I used "gopkg.in/yaml.v3" I got
```
panic: cannot deep copy int [recovered]
        panic: cannot deep copy int
```

Reply:
Thanks for the point to use `sigs.k8s.io/yaml`

Reply:
Done